qa/suites/deepsea/tier4/upgrade/3-test-phase.yaml

```diff
@@ -1,40 +1,115 @@
-        - salt.command:
-                command: 'saltutil.sync_all'
-        - deepsea.orch:
-                state_orch: ceph.maintenance.upgrade
-        - exec:
-                client.salt_master:
-                        - 'ceph --version'
-                        - 'ceph -s'
+- exec:
+        client.salt_master:
+                - 'echo "Salt Master node, pre-upgrade"'
+                - 'cat /etc/os-release'
+- deepsea.install_migration_rpm:
+- deepsea.repository:
+        client.salt_master:
+- deepsea.reboot:
+        client.salt_master:
+            tries: 90
+- deepsea.toolbox:
+        post_upgrade_status_checks:
+            client.salt_master:
+- exec:
+        client.salt_master:
+                - 'deepsea --version || true'
+                - 'salt-run deepsea.version 2>/dev/null || true'
+                - 'salt-run status.report 2>/dev/null'
+- exec:
+        mon.a:
+                - 'ceph --version || true'
+                - 'ceph -s || true'
+- deepsea.toolbox:
+        add_noout:
+                osd.0:
+- exec:
+        osd.0:
+                - 'ceph-volume simple scan --force'
+- deepsea.repository:
+        mon.a:
+- deepsea.reboot:
+        mon.a:
+            tries: 90
+- deepsea.toolbox:
+        post_upgrade_status_checks:
+            mon.a:
+- exec:
+        mon.a:
+                - 'ceph-volume simple activate --all'
+- deepsea.toolbox:
+        rm_noout:
+                osd.0:
+- exec:
+        client.salt_master:
+                - 'salt-run status.report 2>/dev/null'
+- deepsea.toolbox:
+        wait_for_health_ok:
+- exec:
+        client.0:
+                - 'sudo rpm -q nfs-client'
+- deepsea.toolbox:
+        add_noout:
+                osd.1:
+- exec:
+        osd.1:
+                - 'ceph-volume simple scan --force'
+- deepsea.repository:
+        osd.1:
+- deepsea.reboot:
+        osd.1:
+             tries: 90
+- deepsea.toolbox:
+        post_upgrade_status_checks:
+            osd.1:
+- exec:
+        client.0:
+                - 'sudo rpm -q nfs-client'
+- exec:
+        osd.1:
+                - 'ceph-volume simple activate --all'
+- deepsea.toolbox:
+        rm_noout:
+                osd.1:
+- exec:
+        client.salt_master:
+                - 'salt-run status.report 2>/dev/null'
+- deepsea.toolbox:
+        wait_for_health_ok:
+- exec:
+        client.salt_master:
+                - 'ceph osd require-osd-release nautilus'
+                - "salt '*' saltutil.sync_all"
+- deepsea.policy:
+        munge_profile:
+                post_upgrade_policy:
+- deepsea.orch:
+        stage: 1
+- deepsea.orch:
+        stage: 2
+- deepsea.toolbox:
+        remove_openattic:
+- exec:
+        client.salt_master:
+                - "salt '*' grains.delkey restart_igw"
+- deepsea.orch:
+        stage: prep
+- deepsea.orch:
+        stage: 1
+- deepsea.orch:
+        stage: 2
+- deepsea.orch:
+        stage: 3
+- deepsea.orch:
+        stage: 4
```

Comment:
@gekios Does the doc really say to run stages 1 and 2, first, then remove the openattic role and delete the restart_igw grain, and then run stages 0-4?

Reply:
Answer: yes, it does - see https://documentation.suse.com/ses/6/single-html/ses-deployment/#upgrade-main